src/Tinlake.ts

```diff
@@ -30,58 +29,59 @@ const contractNames = [
   'LENDER_DEPLOYER',
   'NFT_FEED',
   'GOVERNANCE',
-]
-
-type AbiOutput = {
-  name: string
-  type: 'address' | 'uint256'
-}
-
-export type EthConfig = {
-  from?: string
-  gasPrice?: string
-  gas?: string
+  'ALLOWANCE_OPERATOR',
+] as const
+
+export type PendingTransaction = {
+  hash?: string
+  status: number
+  error?: string
+  timesOutAt?: number
 }
 
 export type EthersConfig = {
-  provider?: ethers.providers.Provider
+  provider: ethers.providers.Provider
   signer?: ethers.Signer
 }
 
-export type ContractNames = typeof contractNames[number]
+export type ContractName = typeof contractNames[number]
 
 export type Contracts = {
-  [key in ContractNames]?: any
+  [key in ContractName]?: any
 }
 
 export type ContractAbis = {
-  [key in ContractNames]?: any
+  [key in ContractName]?: any
 }
 
 export type ContractAddresses = {
-  [key in ContractNames]?: string
+  [key in ContractName]?: string
 }
 
 export type TinlakeParams = {
-  provider: any
+  ethersConfig: EthersConfig
```

Comment:
Good point! I originally added this because we also had `ethConfig` and I was running them side by side. But it doesn't make sense anymore. I'll replace it with top level provider and signer properties, good idea.